cmake/openssl_functions.cmake

```diff
@@ -44,7 +44,8 @@ function (openssl_set_defines)
     SSL_CTX_set_num_tickets
     SSL_CTX_set_ciphersuites
     SSL_CTX_set1_sigalgs_list
-    SSL_CTX_set1_client_sigalgs_list)
+    SSL_CTX_set1_client_sigalgs_list
+    SSL_CONF_cmd)
```

Comment:
I think we could require libssl 1.0.2 where this function is already available, so we wouldn't need to check specifically for this function and we won't need to make it conditional in the codebase.

Right now, configure.ac lists 0.9.8 as minimum, and cmake does not seem to declare any version dependency, but I think it's a mistake to build syslog-ng against anything below 1.0.2 (which is centos-7's version). Honestly I feel more of requiring 1.1.0 but centos-7 is still an important platform for us.

Reply:
MrAnno had a very close opinion on this, maybe it is time to set libssl 1.0.2 as the bare minimum?!

Reply:
Yes, please do so, and eliminate any conditional code (not just the ones related to this PR but the existing ones too) in the process.

Reply:
I've found the following probably related conditionals

    #cmakedefine01 SYSLOG_NG_HAVE_DECL_EVP_MD_CTX_RESET
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_ASN1_STRING_GET0_DATA
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_SSL_CTX_GET0_PARAM
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_SSL_CTX_SET_CIPHERSUITES
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_SSL_CTX_SET1_SIGALGS_LIST
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_SSL_CTX_SET1_CLIENT_SIGALGS_LIST
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_SSL_CTX_SET_NUM_TICKETS
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_X509_STORE_CTX_GET0_CERT
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_X509_GET_EXTENSION_FLAGS
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_DH_SET0_PQG
    #cmakedefine01 SYSLOG_NG_HAVE_DECL_BN_GET_RFC3526_PRIME_2048

Unfortunately, in some cases, I cannot find in the documentation which version the given functionality is introduced.

Do you mean everywhere these can be found?

Reply:
I am fine with just bumping the requirement to 1.0.2 and not checking for SSL_CONF_cmd, but assume it's there.

The existing ones can eliminated later or now, as you prefer.

Reply:
I've already removed the new conditional introduced with my changes and I prefer to add that refactoring to a separate run instead, thank you!